controls/build.gradle

```diff
@@ -1,19 +1,88 @@
+import java.text.SimpleDateFormat
 import com.vanniktech.maven.publish.SonatypeHost
 import com.vanniktech.maven.publish.JavaLibrary
 import com.vanniktech.maven.publish.JavadocJar
 
 plugins {
-	// Apply the java-library plugin for API and implementation separation.
-	id 'java-library'
+	id "java"
+	id "maven-publish"
+	id "com.peterabeles.gversion" version "1.10"
 	id "com.diffplug.spotless" version "6.24.0"
 	id "com.vanniktech.maven.publish" version "0.30.0"
 }
 
+java {
+	sourceCompatibility = JavaVersion.VERSION_17
+	targetCompatibility = JavaVersion.VERSION_17
+}
+
+// Set this to true to enable desktop support.
+def includeDesktopSupport = true
+
+configurations.all {
+	// exclude group: "edu.wpi.first.wpilibj"
+}
+
+task(checkAkitInstall, dependsOn: "classes", type: JavaExec) {
+	mainClass = "org.littletonrobotics.junction.CheckInstall"
+	classpath = sourceSets.main.runtimeClasspath
+}
+// TODO: Figure out why advantagekit won't let us build the project
+//compileJava.finalizedBy checkAkitInstall
+
+// Defining my dependencies. In this case, WPILib (+ friends), and vendor libraries.
+// Also defines JUnit 5.
+dependencies {
+
+	//def akitJson = new groovy.json.JsonSlurper().parseText(new File(projectDir.getAbsolutePath() + "/vendordeps/AdvantageKit.json").text)
+	//annotationProcessor "org.littletonrobotics.akit.junction:junction-autolog:$akitJson.version"
+	def akitVersion = "4.0.0-alpha-1"
+	annotationProcessor "org.littletonrobotics.akit.junction:junction-autolog:$akitVersion"
+
+	implementation "org.littletonrobotics.akit.junction:junction-core:$akitVersion"
```

Comment:
@linglejack06 this seems to be working for @minhnguyenbhs - maybe worth a look? Idk what the difference is off the top of my head though